server/src/services/formatting/forgeFormat.ts

```diff
@@ -19,34 +22,32 @@ export async function forgeFormat(
   logger.trace(`Forge command: ${forgeCommand}`);
   logger.trace(`CWD: ${cwd}`);
 
-  // Spawn and promisify the forge fmt process
-  const execPromise = promisify(cp.exec)(`${forgeCommand} fmt --raw -`, {
-    cwd,
-  });
-
-  // Write the file contents to the process
-  const { child } = execPromise;
-  child.stdin?.write(text);
-  child.stdin?.end();
-
-  // Set a time limit
-  const timeout = setTimeout(() => {
-    child.kill("SIGKILL");
-  }, 2000);
-
   // Wait for the formatted output
-  const formattedText = (await execPromise).stdout;
-
-  clearTimeout(timeout);
-
-  // Build and return a text edit with the entire file content
-  const textEdit: TextEdit = {
-    range: {
-      start: { line: 0, character: 0 },
-      end: document.positionAt(text.length),
-    },
-    newText: formattedText,
-  };
-
-  return [textEdit];
+  try {
+    const { stdout: formattedText } = await execWithInput(
+      `${forgeCommand} fmt --raw -`,
+      text,
+      {
+        cwd,
+        timeout: FORMAT_TIMEOUT,
+      }
+    );
+
+    // Build and return a text edit with the entire file content
+    const textEdit: TextEdit = {
+      range: {
+        start: { line: 0, character: 0 },
+        end: document.positionAt(text.length),
+      },
+      newText: formattedText,
+    };
+
+    return [textEdit];
+  } catch (error: any) {
+    if (error.killed) {
+      throw new TimeoutError(FORMAT_TIMEOUT);
+    } else {
+      throw error;
+    }
+  }
```

Comment:
```suggestion
  } catch (error: unknown) {
    if (error instanceof Error && "killed" in error && error.killed) {
      throw new TimeoutError(FORMAT_TIMEOUT);
    } else {
      throw error;
    }
  }
```
We are trying to avoid `any`, I suspect we will want to add helpers to utils to better support error detection patterns.

Reply:
Good point. I just stumbled upon the "ensureError" logic on hardhat. We could use something like that in the future